app/src/editor/canvas/components/ModuleHelp.jsx

```diff
@@ -22,23 +26,53 @@ export default class ModuleHelp extends React.Component {
     }
 
     async load() {
-        const { moduleId } = this.props
-        const help = await services.moduleHelp({
-            id: moduleId,
-        })
+        const { moduleId, moduleName } = this.props
+
+        const cleanedName = moduleName.replace(/\s/g, '').replace(/\(/g, '_').replace(/\)/g, '')
+
+        // eslint-disable-next-line global-require, import/no-dynamic-require
+        const helpJson = await require(`$newdocs/content/canvasModules/${cleanedName}-${moduleId}.jsx`)
+        const { help } = helpJson.default
+
+        // eslint-disable-next-line global-require, import/no-dynamic-require
+        const helpMdPath = await require(`$newdocs/content/canvasModules/${cleanedName}-${moduleId}.md`)
+        const helpMdText = await fetch(helpMdPath).then((res) => res.text())
+
```

Comment:
What happens if user changes the current help and the old help requests arrive before the new help requests?

A simple thing you could do is to check whether `this.props.moduleId` still equals `moduleId`, if not then stop.

```js
if (this.props.moduleId !== moduleId) { return }
```

This is one reason the state was originally keyed by `moduleId`, doesn't matter which order the results arrive.

Reply:
@fonty1 other stuff just nitpicking, this I think is a blocker

Reply:
Another option is setting the `key` in some wrapper to `moduleId`, when `moduleId` changes, the component is unmounted and state cleared